x-pack/plugins/apm/common/agent_configuration/setting_definitions/index.test.ts

```diff
@@ -43,134 +43,150 @@ describe('filterByAgent', () => {
 
   describe('options per agent', () => {
     it('go', () => {
-      expect(getSettingKeysForAgent('go')).toEqual([
-        'capture_body',
-        'capture_headers',
-        'log_level',
-        'recording',
-        'sanitize_field_names',
-        'span_frames_min_duration',
-        'stack_trace_limit',
-        'transaction_ignore_urls',
-        'transaction_max_spans',
-        'transaction_sample_rate',
-      ]);
+      expect(getSettingKeysForAgent('go')).toEqual(
+        expect.arrayContaining([
```

Comment:
Not sure whether this change is controversial. The test is now passing if `getSettingKeysForAgent` returns more elements. This is to make it easier for agent sub-teams to just add their agent name in `includeAgents` directly from, without having to clone the repo and set up Kibana.

Reply:
Hmm.. it's a little controversial in the sense that it's no reflected accurately which settings are available for a particular agent.

Reply:
... but if this makes it much easier for agent devs I think it's ok. They are going to me the primary audience of this.

Reply:
To me, it feels a bit redundant having to add an agent in `includeAgents` and having to update the test at the same time. So the `expect.arrayContaining` feels like a good compromise where we'll at least make sure that existing options don't disappear without notice. But definitely a debatable change.